parser.hh

```diff
@@ -388,6 +412,9 @@ struct ParserPosition
 	///iterator into the input.
 	Input::iterator it;
 
+	///user-meaningful filename.
+	std::string filename() const { return it.input().name(); }
```

Comment:
Should probably be returning a `const` reference here too and save a copy unless the caller actually wants it.